vector/src/main/java/im/vector/app/features/home/room/detail/composer/MessageComposerViewModel.kt

```diff
@@ -817,25 +817,26 @@ class MessageComposerViewModel @AssistedInject constructor(
         }
     }
 
-    private fun handleEndRecordingVoiceMessage(isCancelled: Boolean, rootThreadEventId: String? = null) {
-        audioMessageHelper.stopPlayback()
-        if (isCancelled) {
-            audioMessageHelper.deleteRecording()
-        } else {
-            audioMessageHelper.stopRecording(convertForSending = true)?.let { audioType ->
-                if (audioType.duration > 1000) {
-                    room.sendMedia(
-                            attachment = audioType.toContentAttachmentData(isVoiceMessage = true),
-                            compressBeforeSending = false,
-                            roomIds = emptySet(),
-                            rootThreadEventId = rootThreadEventId)
-                } else {
+    private fun handleEndRecordingVoiceMessage(isCancelled: Boolean, rootThreadEventId: String? = null) =
+            viewModelScope.launch {
```

Comment:
Keep in mind that` viewModelScope.launch` default thread is the Main. If we want to run it on IO for example we should do something like this` viewModelScope.launch(Dispatchers.IO)`